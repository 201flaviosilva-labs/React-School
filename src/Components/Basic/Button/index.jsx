import React from "react";

export default function Button(props) {
	return (
		<button onClick={() => props.children == "-" ? props.setN(n - 1) : props.setN(n + 1)}>{props.n}{props.children == "-" ? "-" : "+"}1</button>
	)
}
