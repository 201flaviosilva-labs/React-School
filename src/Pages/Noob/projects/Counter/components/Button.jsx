import React from "react";

export default function Button({ onClick, label }) {
	return (
		<button data-testid="counter-button" onClick={onClick}>
			<span>{label}</span>
		</button>
	);
}
