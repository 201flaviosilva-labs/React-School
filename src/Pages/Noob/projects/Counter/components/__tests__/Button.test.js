import React from "react";
import { screen, render, cleanup } from "@testing-library/react";

import Button from "../Button";

beforeEach(cleanup);
afterEach(cleanup);

it("Render without crashing", () => {
	const { asFragment } = render(<Button />);
	expect(asFragment()).toMatchSnapshot();
});

it("Render button correctly", () => {
	const onClick = () => {
		console.log("Click");
	};
	render(<Button onClick={onClick} label={"Some Text"} />);
	const button = screen.getByTestId("counter-button");
	expect(button).toBeInTheDocument();
	expect(button).toHaveTextContent("Some Text");
	expect(button).toContainHTML("<span>Some Text</span>");
});
