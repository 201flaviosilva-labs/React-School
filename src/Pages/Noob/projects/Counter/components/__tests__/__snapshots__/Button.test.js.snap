// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without crashing 1`] = `
<DocumentFragment>
  <button
    data-testid="counter-button"
  >
    <span />
  </button>
</DocumentFragment>
`;
